Controller/ExceptionController.php

```diff
@@ -26,8 +27,34 @@
 /**
  * Custom ExceptionController that uses the view layer and supports HTTP response status code mapping.
  */
-class ExceptionController extends ContainerAware
+class ExceptionController
 {
+    private $exceptionWrapperHandler;
+    private $formatNegotiator;
+    private $viewHandler;
+    private $templating;
+    private $exceptionCodes;
+    private $exceptionMessages;
+    private $showException;
+
+    public function __construct(
+        ExceptionWrapperHandlerInterface $exceptionWrapperHandler,
+        FormatNegotiator $formatNegotiator,
+        ViewHandlerInterface $viewHandler,
+        EngineInterface $templating,
```

Comment:
making the service lazy would be hard (as you don't own the service, and it would add overhead to anyone using it all the time because of the proxy wrapper).

However, there is something else needed here: 

- it should be possible to build this service when there is no templating system in the project (assuming you don't use the HTML format)
- I would suggest using Twig directly instead of the templating component wrapping twig, as you only render Twig templates anyway